### PR TITLE
Nucleo Wl55JC board supports E-ink display

### PR DESCRIPTION
**Pull request by adamkaliszan**
## Summary
Stm NucleoWl55 BSP for E-ink display

BSP suooprt SPI driver. 
Make menuconfig allow to chose pins that are conected to external e-ink display with ssd1680 driver.  

## Impact
None

## Testing
Tested on Stm Nucleo Wl55JC

### CODE REVIEW

File: boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_spi.c [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+#ifdef CONFIG_STM32WL5_SPI1
+  /* Configure SPI-based devices */
+
+  g_spi1 = stm32wl5_spibus_initialize(1);
+  if (!g_spi1)
+    {
+      spierr("ERROR: FAILED to initialize SPI port 1\n");
+    }
+
+  g_spi1 = stm32wl5_spibus_initialize(1);
```
Comment (mlyszczek) [inline boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_spi.c:80]: Duplicated code? If not maybe add a comment that you try again to init?

File: boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_spi.c [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  if (!g_spi1)
+    {
+      spierr("ERROR: FAILED to initialize SPI port 1\n");
```
Comment (mlyszczek) [inline boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_spi.c:83]: Does it even make sense to move on when spi fails to initialize?

File: boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_ssd1680.c [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+static struct lcd_dev_s    *g_lcddev;
+struct ssd1680_priv_s g_ssd1680_priv =
+{
+#if defined(CONFIG_SSD1680_GPIO_PIN_PWR) && (CONFIG_SSD1680_GPIO_PIN_PWR >= 0)
+  .set_vcc = ssd1680_set_vcc,
+#else
+  .set_vcc = NULL,
+#endif
+#if defined(CONFIG_SSD1680_GPIO_PIN_RST)  && (CONFIG_SSD1680_GPIO_PIN_RST >= 0)
```
Comment (mlyszczek) [inline boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_ssd1680.c:84]: Wouldn't it be more readable with empty line between the lines?

File: boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_spi.c (hunk 2 above)
Comment (adamkaliszan) [reply boards/arm/stm32wl5/nucleo-wl55jc/src/stm32_spi.c:83]: Initialization of additional lines for SPI devices is pointless.

**Pull request merged**